src/rust/bitbox02-rust/src/async_usb.rs

```diff
@@ -160,6 +163,11 @@ pub fn spin() {
                 {
                     *task = popped_task;
                 } else {
+                    // Happens if the ongoing task is cancelled but execution resumes to the a yield
+                    // point, which can only really happen if you call `cancel()` inside the task
+                    // (inside the Rust async fn), but then advance to another `.await` instead of
+                    // returning. (An automatic cancel due to a timeout happens while task is not
+                    // executing, which is why in that case we don't run into this).
```

Comment:
For the record, an alternative is to simply allow this, see https://github.com/BitBoxSwiss/bitbox02-firmware/pull/1433. I'll close that PR though b/c we don't actually need this.